File: seed/pydantic-v2/pagination/resources/users/types/user_optional_list_container.py
from pydantic import BaseModel
from typing import Optional, List
from resources.users.types.user import User
from dt import datetime
from core.datetime_utils import serialize_datetime


class UserOptionalListContainer(BaseModel):
    users: Optional[List[User]] = None

    class Config:
        frozen = True
        smart_union = True
        json_encoders = {datetime: serialize_datetime}

File: seed/ruby-sdk-v2/pagination/lib/fern_pagination/users/types/user_optional_list_container.rb
# frozen_string_literal: true

module FernPagination
  module Users
    module Types
      class UserOptionalListContainer < Internal::Types::Model
        field :users, -> { Internal::Types::Array[FernPagination::Users::Types::User] }, optional: true, nullable: false
      end
    end
  end
end
